### ISSUE TITLE
404 from Xero results in parsing error

### ISSUE THREAD

**Issue by GuiSim** [XeroAPI/Xero-Java#32]
Call `getAccount("123")`.

If the account does not exist, you will receive an error similar to this:

```
Caused by: java.lang.IllegalStateException: Error unmarshalling response: The resource you're looking for cannot be found
	at com.xero.api.XeroClient.unmarshallResponse(XeroClient.java:366)
	at io.pleo.xero.internal.ExtendedXeroClient.newApiException(ExtendedXeroClient.java:55)
	at com.xero.api.XeroClient.get(XeroClient.java:112)
	at com.xero.api.XeroClient.get(XeroClient.java:87)
	at com.xero.api.XeroClient.getAccount(XeroClient.java:424)
	at io.pleo.xero.XeroService.createAccount(XeroService.java:67)
	... 70 common frames omitted
Caused by: javax.xml.bind.UnmarshalException: null
	at javax.xml.bind.helpers.AbstractUnmarshallerImpl.createUnmarshalException(AbstractUnmarshallerImpl.java:335)
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.createUnmarshalException(UnmarshallerImpl.java:578)
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:264)
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:236)
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:291)
	at com.xero.api.XeroClient.unmarshallResponse(XeroClient.java:364)
	... 75 common frames omitted
Caused by: org.xml.sax.SAXParseException: Content is not allowed in prolog.
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:203)
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:177)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
	at com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1472)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:994)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:602)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:505)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:841)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:770)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1213)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:643)
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:258)
``` 

It seems that the recent rework of exception parsing has broken this behavior.

I would have expected some sort of `NotFoundException` or more specific.

**Comment by SidneyAllen** [comment]
We got a request for better error handling - so I've added code handle  (400, 404, 503, etc) using a  try catch block - when an error is encountered an "XeroApiExecption' is thrown returning the response code and message.

This example is forcing a 404 

try {
	Contact ContactOne = client.getContact("1234");
} catch (XeroApiException e) {
	System.out.println(e.getResponseCode());
	System.out.println(e.getMessage());	
}

**Comment by GuiSim** [comment]
@SidneyAllen you can look at the stacktrace above.

`getAccount` throws `IllegalStateException`, not `XeroApiException`.

Great job on the rework of the exception mechanism. I love having tightly typed exceptions. It just seems that in certain scenarios, the Xero API doesn't return XML but instead returns text.

Here, the text is `"The resource you're looking for cannot be found"`.

I would like this issue to be reopened.

**Comment by SidneyAllen** [comment]
@GuiSim - thanks.

While adding this exception handling I found the same thing.  For 400's (validation errors) we return XML.  For 404 we return text and for 503 we return key/value pairs separated by "&".  In each of these cases I've added code to parse the response and throw them to the catch block.  Take a look in at [XeroClient](https://github.com/XeroAPI/Xero-Java/blob/master/src/main/java/com/xero/api/XeroClient.java) at the XeroApiException method.

Are you using version 0.4.1 of the SDK?  When I test it and force a 404, the "text" is handled correctly and passed back as a message threw the XeroApiException.
